packages/did-provider-ebsi/src/EbsiDidProvider.ts

```diff
@@ -120,11 +128,69 @@ export class EbsiDidProvider extends AbstractIdentifierProvider {
     throw Error(`Not (yet) implemented for the EBSI did provider`)
   }
 
+  private assertedKey = (args: { key?: IKeyOpts; type: EbsiKeyType; kms?: string }): MinimalImportableKey => {
+    const { key, type, kms } = args
+    const minimalImportableKey: Partial<MinimalImportableKey> = { ...key } ?? {}
+    minimalImportableKey.kms = this.assertedKms(kms)
+    minimalImportableKey.type = this.assertedKeyType({ key, type })
+    minimalImportableKey.meta = { purposes: this.assertedPurposes({ key, type }) }
+    return minimalImportableKey as MinimalImportableKey
+  }
+
   private assertedKms(kms?: string) {
     const result = kms ?? this.defaultKms
     if (!!result) {
       return result
     }
     throw Error('no KMS supplied')
   }
+
+  private assertedKeyType = (args: { key?: IKeyOpts; type: EbsiKeyType }): EbsiKeyType => {
+    if (!args.key?.type) {
+      return args.type
+    }
+    return args.key.type
+  }
+
+  private assertedPurposes = (args: { key?: IKeyOpts; type: EbsiKeyType }) => {
+    const { key, type } = args
+    if (key?.purposes && key.purposes.length > 0) {
+      switch (key.type) {
+        case 'Secp256k1': {
+          if (key?.purposes && key.purposes.length > 0 && key.purposes?.includes(EbsiPublicKeyPurpose.CapabilityInvocation)) {
+            return key.purposes
+          }
+          throw new Error(`Secp256k1 key requires ${EbsiPublicKeyPurpose.CapabilityInvocation} purpose`)
+        }
+        case 'Secp256r1': {
+          if (
+            key?.purposes &&
+            key.purposes.length > 0 &&
+            key.purposes.every((purpose) => [EbsiPublicKeyPurpose.AssertionMethod, EbsiPublicKeyPurpose.Authentication].includes(purpose))
+          ) {
+            return key.purposes
+          }
+          throw new Error(`Secp256r1 key requires ${[EbsiPublicKeyPurpose.AssertionMethod, EbsiPublicKeyPurpose.Authentication].join(', ')} purposes`)
+        }
+        default:
+          throw new Error(`Unsupported key type: ${key.type}`)
+      }
+    }
+    return this.setDefaultPurposes({ key, type })
```

Comment:
Hmzz this is rather ugly. An assert method setting defaults instead of sticking to assertions. Either rename the method slightly or adjust the implementation